CODING_GUIDELINES.md

```diff
@@ -81,6 +88,12 @@ Try to keep sections and items logically ordered, add section separator comments
 
 Consider configuration files as another type of source code and apply the same criteria, using comments when possible for better readability.
 
+### Import conventions
+
+- Library code should import the defining module of a code object, unless a proxy is provided specifically for some reason
+- Test, doctest and example code should import the designated public API module (if exists) and use code objects from there
+- Test, doctest and example code should import public API objects from the designated public API module (if exists) if they are used in DSL code.
```

Comment:
This is fine for me.

Another option which could help to make it slightly shorter and easier to remember is to rename the `common questions` subsection above to `other questions` or `additional questions` (or something like that) and add it as a new point there (e.g. `importing symbols reexported in different locations`) mentioning the different import strategy for library code vs user code (test, doctests and examples)